cin_validator/rules/cin2022_23/rule_8696.py

```diff
@@ -0,0 +1,96 @@
+from typing import Mapping
+
+import pandas as pd
+
+from cin_validator.rule_engine import (
+    CINTable,
+    IssueLocator,
+    RuleContext,
+    rule_definition,
+)
+from cin_validator.test_engine import run_rule
+from cin_validator.utils import make_census_period
+
+# Get tables and columns of interest from the CINTable object defined in rule_engine/__api.py
+# Replace ChildIdentifiers with the table name, and LAChildID with the column name you want.
+
+Assessments = CINTable.Assessments
+Header = CINTable.Header
+AssessmentAuthorisationDate = Assessments.AssessmentAuthorisationDate
+ReferenceDate = Header.ReferenceDate
+
+# define characteristics of rule
+@rule_definition(
+    code=8696,
+    module=CINTable.Assessments,
+    message="Assessment end date must fall within the census year",
+    affected_fields=[AssessmentAuthorisationDate, ReferenceDate],
+)
+def validate(
+    data_container: Mapping[CINTable, pd.DataFrame], rule_context: RuleContext
+):
+    # Replace ChildIdentifiers with the name of the table you need.
+    df = data_container[Assessments]
+    df_ref = data_container[Header]
+
+    ref_data_series = df_ref[ReferenceDate]
+    collection_start, collection_end = make_census_period(ref_data_series)
+
+    # implement rule logic as described by the Github issue. Put the description as a comment above the implementation as shown.
+
+    # <If present <AssessmentAuthorisationDate> (N00160) must be on or between [Start_Of_Census_Year] and <ReferenceDate> (N00603)
+
+    df = df[
+       (
+            (df[AssessmentAuthorisationDate] < collection_start)
+            | (df[AssessmentAuthorisationDate] > collection_end)
+       )
+    ]
+
+    failing_indices = df.index
+
+    rule_context.push_issue(
+        table=Assessments, field=AssessmentAuthorisationDate, row=failing_indices
+    )
+
+    # Replace ChildIdentifiers and LAchildID with the table and column name concerned in your rule, respectively.
+    # If there are multiple columns or table, make this sentence multiple times.
+
+
+def test_validate():
+    # Create some sample data such that some values pass the validation and some fail.
+
+    miss_auth = pd.to_datetime(
+        [
+            "01/03/2019",
+            "01/04/2021",
+            "01/10/2022",
```

Comment:
Hi @tab1tha 

I'm happy to do the final touches to the code so that there is less for you to do. I'll now select another rule from the list and when doing that try and conclude the whole process including the final touches.

Thanks,
Gavin

Reply:
That's kind of you, @GBurrel 

Don't worry about it. The formatting process is automatic so that is done quite quickly. When next we meet, do tell me to show you how to run the formatter too !. (or you can ask Will).

If doing the changes helps you remember what was done, I can tag you to the changes but if you're busy with work at that time, you can also tell me to complete it. It's a team effort :)